src/data/template/static/js/control/mapselect.js

```diff
@@ -23,7 +26,10 @@ MapSelectControl.prototype.create = function(wrapper) {
 
 		link.addEventListener("click", (function(ui) {
 			return function(a) {
-				ui.setMap(a.target.getAttribute('data-world'));
+				var h = a.target.getAttribute('data-world');
+				var button = document.getElementById('control-wrapper-map-select').getElementsByTagName('button')[0];
+				button.innerHTML = this.ui.getMapConfig(h).name + ' <span class="caret"></span>';
```

Comment:
Ah yeah, you're probably right. It seemed to work in basic testing but `this` should be scoped to the `link` element.